src/components/Drawer.js

```diff
@@ -0,0 +1,146 @@
+const React = require('react');
+const Velocity = require('velocity-animate');
+
+const StyleConstants = require('../constants/Style');
+
+const Icon = require('../components/Icon');
+
+const Drawer = React.createClass({
+  propTypes: {
+    duration: React.PropTypes.number,
+    easing: React.PropTypes.array,
+    isOpen: React.PropTypes.bool,
+    onClose: React.PropTypes.func.isRequired,
+    siblingContent: React.PropTypes.shape({
+      currentSiblingIndex: React.PropTypes.number.isRequired,
+      nextSibling: React.PropTypes.func.isRequired,
+      previousSibling: React.PropTypes.func.isRequired,
+      totalSiblings: React.PropTypes.number.isRequired
+    })
+  },
+
+  getDefaultProps () {
+    return {
+      duration: 500,
+      isOpen: false,
+      easing: [0.28, 0.14, 0.34, 1.04]
+    };
+  },
+
+  componentDidMount () {
+    this._renderTransition(this.props.isOpen);
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (newProps.isOpen !== this.props.isOpen) {
+      this._renderTransition(newProps.isOpen);
+    }
+  },
+
+  _renderSiblingContent () {
+    const styles = this.styles();
+
+    return (
+      <div ref={(ref) => (this._siblingContent = ref)} style={styles.siblingContent}>
```

Comment:
@cerinman https://facebook.github.io/react/docs/more-about-refs.html here's the react docs on that:

https://facebook.github.io/react/docs/more-about-refs.html